lib/ansible/modules/cloud/amazon/ec2_vpc_endpoint.py

```diff
@@ -0,0 +1,347 @@
+        else:
+            module.fail_json(msg=str(e))
+    except Exception as e:
+        module.fail_json(msg=str(e))
+
+    return changed, result
+
+
+def setup_removal(client, module):
+    params = dict()
+    changed = False
+    params['DryRun'] = module.check_mode
+    if not module.params.get('vpc_endpoint_id'):
+        module.fail_json(msg='vpc_endpoint_id is a required paramater')
+    else:
+        if isinstance(module.params.get('vpc_endpoint_id'), basestring):
+            params['VpcEndpointIds'] = [module.params.get('vpc_endpoint_id')]
+        else:
+            params['VpcEndpointIds'] = module.params.get('vpc_endpoint_id')
+    try:
+        result = client.delete_vpc_endpoints(**params)['Unsuccessful']
+        if not module.check_mode and (result != []):
+            module.fail_json(msg=result)
+    except botocore.exceptions.ClientError as e:
+        if "DryRunOperation" in e.message:
+            changed = True
+            result = 'Would have deleted VPC Endpoint if not in check mode'
+        else:
+            module.fail_json(msg=str(e))
+    except Exception as e:
+        module.fail_json(msg=str(e))
+
+    return changed, result
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            vpc_id=dict(),
+            service=dict(),
+            policy=dict(),
+            state=dict(default='present', choices=['present', 'absent']),
+            wait=dict(type='bool', default=False),
+            wait_timeout=dict(type='int', default=320, required=False),
+            route_table_ids=dict(type='list'),
+            vpc_endpoint_id=dict(),
+            client_token=dict(),
+        )
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    # Validate Requirements
+    if not HAS_BOTO3:
+        module.fail_json(msg='botocore and boto3 are required for removal')
+
+    state = module.params.get('state')
+
+    try:
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    except NameError as e:
+        # Getting around the get_aws_connection_info boto reliance for region
+        if "global name 'boto' is not defined" in e.message:
+            module.params['region'] = botocore.session.get_session().get_config_variable('region')
+            if not module.params['region']:
+                module.fail_json(msg="Error - no region provided")
+        else:
+            module.fail_json(msg="Can't retrieve connection information - " + str(e))
+
+    try:
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+        ec2 = boto3_conn(module, conn_type='client', resource='ec2', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except botocore.exceptions.NoCredentialsError as e:
+        module.fail_json(msg=str(e))
```

Comment:
Please add a message like "Failed to connect to AWS due to wrong or missing credentials" rather than just `str(e)`